src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.Data.cs

```diff
@@ -0,0 +1,104 @@
+            MinimumOSPlatformAttribute,
+            ObsoletedInOSPlatformAttribute,
+            RemovedInOSPlatformAttribute,
+        }
+
+        private readonly struct PlatformAttributeInfo : IEquatable<PlatformAttributeInfo>
+        {
+            public PlatformAttributeType AttributeType { get; }
+            public string PlatformName { get; }
+            public Version Version { get; }
+
+            private PlatformAttributeInfo(PlatformAttributeType attributeType, string platformName, Version version)
+            {
+                AttributeType = attributeType;
+                PlatformName = platformName;
+                Version = version;
+            }
+
+            public static bool TryParsePlatformAttributeInfo(AttributeData osAttribute, out PlatformAttributeInfo parsedAttribute)
+            {
+                if (!osAttribute.ConstructorArguments.IsEmpty &&
+                    osAttribute.ConstructorArguments[0] is { } argument &&
+                    argument.Kind == TypedConstantKind.Primitive &&
+                    argument.Type.SpecialType == SpecialType.System_String &&
+                    !argument.IsNull &&
+                    !argument.Value.Equals(string.Empty) &&
+                    TryParsePlatformNameAndVersion(osAttribute.ConstructorArguments[0].Value.ToString(), out string platformName, out Version? version))
+                {
+                    parsedAttribute = new PlatformAttributeInfo(SwitchAttrributeType(osAttribute.AttributeClass.Name), platformName, version);
+                    return true;
+                }
+                parsedAttribute = default;
+                return false;
+            }
+
+            private static PlatformAttributeType SwitchAttrributeType(string osAttributeName)
+                => osAttributeName switch
+                {
+                    MinimumOSPlatformAttribute => PlatformAttributeType.MinimumOSPlatformAttribute,
+                    ObsoletedInOSPlatformAttribute => PlatformAttributeType.ObsoletedInOSPlatformAttribute,
+                    RemovedInOSPlatformAttribute => PlatformAttributeType.RemovedInOSPlatformAttribute,
+                    TargetPlatformAttribute => PlatformAttributeType.TargetPlatformAttribute,
+                    _ => throw new NotImplementedException(),
+                };
+
+            public override bool Equals(object obj)
+            {
+                if (obj is PlatformAttributeInfo info)
+                {
+                    return Equals(info);
+                }
+                return false;
+            }
+
+            public override int GetHashCode() => HashUtilities.Combine(AttributeType.GetHashCode(), PlatformName.GetHashCode(), Version.GetHashCode());
+
+            public static bool operator ==(PlatformAttributeInfo left, PlatformAttributeInfo right) => left.Equals(right);
+
+            public static bool operator !=(PlatformAttributeInfo left, PlatformAttributeInfo right) => !(left == right);
```

Comment:
What if instead of invoking the `==` operator method from the previous line, you directly invoke ` !left.Equals(right)`? You save one invocation.

Reply:
removed by new design